fews_3di/simulation.py

```diff
@@ -326,6 +328,27 @@ def _add_initial_state(self, saved_state_id_file: Path):
                 if self.allow_missing_saved_state:
                     logger.warn(msg)
                     return
+                elif cold_state_id_file.exists():
+                    msg = f"Warm state failed to set; Cold state id file {cold_state_id_file} found"
+                    logger.info(msg)
+                    cold_state_id: str = cold_state_id_file.read_text().strip()
+                    logger.info("Simulation will use initial state %s", cold_state_id)
+                    try:
+                        self.simulations_api.simulations_initial_saved_state_create(
+                            self.simulation_id, data={"saved_state": cold_state_id}
+                        )
+                        return
+                    except openapi_client.exceptions.ApiException as f:
+                        if f.status == 400:
+                            logger.debug("Cold state setting error: %s", str(f))
+                            msg = (
+                                f"Setting initial state to cold state id={cold_state_id} failed. "
+                                f"The error response was {f.body}"
+                            )
+                            raise MissingSavedStateError(msg) from e
+                        else:
+                            logger.debug("Error isn't a 400, so we re-raise it.")
+                            raise
```

Comment:
Hier zit heel veel duplicatie in met de code hierboven.
Kan je niet zoiets doen?

    for state_file in [saved_state_file, cold_state_file]:
        if not_exist:
            continue
        try/except openapiclient, return als het lukt
    if not allow_missing:
        raise MissingSavedStateError